iceoryx_examples/iceperf/iceperf_follower.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019 - 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2020 - 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceperf_follower.hpp"
+#include "iceoryx.hpp"
+#include "iceoryx_c.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "mq.hpp"
+#include "topic_data.hpp"
+#include "uds.hpp"
+
+#include <iostream>
+
+//! [use constants instead of magic values]
+constexpr const char APP_NAME[]{"iceperf-bench-follower"};
+constexpr const char PUBLISHER[]{"Laurel"};
+constexpr const char SUBSCRIBER[]{"Hardy"};
+//! [use constants instead of magic values]
+
+//! [do the measurement for a single technology]
+void IcePerfFollower::doMeasurement(IcePerfBase& ipcTechnology) noexcept
+{
+    ipcTechnology.initFollower();
+
+    ipcTechnology.latencyPerfTestFollower();
+
+    ipcTechnology.shutdown();
+}
+//! [do the measurement for a single technology]
+
+//! [get the settings for the performance measurement]
+PerfSettings IcePerfFollower::getSettings(iox::popo::Subscriber<PerfSettings>& subscriber) noexcept
+{
+    // wait for settings from leader application
+    constexpr bool WAIT_FOR_SETTINGS{true};
+    while (WAIT_FOR_SETTINGS)
+    {
+        static bool waitMessagePrinted{false};
+        if (!waitMessagePrinted)
+        {
+            std::cout << "Waiting for PerfSettings from leader application!" << std::endl;
+            waitMessagePrinted = true;
+        }
+
+        auto sample = subscriber.take();
+        if (sample)
+        {
+            return *(sample.value());
+        }
+        else
+        {
+            constexpr std::chrono::milliseconds POLLING_INTERVAL{100};
```

Comment:
I prefer all the "settings" of a method at the top of the method so that you can see with one glimpse on which screws you can screw when you adjust the test.
If you think so too could you please move `POLLING_INTERVAL` at the top?

Reply:
This is less of a setting and more of giving a magic number a name. In general I would agree with you but in this case it just distributes two things at different places which actually belong together.